### PR TITLE
BaseDistribution: Add naive .plot()

### LINKED ISSUE #4: Add `.plot()` to `BaseDistribution`
Instead of asking the user (forecaster) to explicitly create `samples` and use them for plotting, I suggest adding `.plot()` to each distribution.

My main argument for this being "correct" is that [pandas DataFrame has a `plot` method](https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.plot.html).

Also, here's some code from the readme and how I imagine it changing if we had `.plot()`:

Current code:

```
prior = sq.norm(1,5)
prior_samples = prior @ (10*K)
plt.hist(prior_samples, bins = 200)
plt.show()
```

My suggestion:

```
prior = sq.norm(1,5)
prior.plot() # optionally add `bins` and maybe other arguments
```

 Do you agree this would be a good idea or do you have some pushback maybe?

### PR DESCRIPTION
**Pull request by hibukki**
Nothing fancy, just replace

```
samples = total_tuners_in_2022 @ 1000
plt.hist(samples, bins=200)
plt.show()
```

with

```
total_tuners_in_2022.plot()
```

I think more can be done here, like using Seaborn, but I tried keeping this PR simple and uncontroversial.

The old functionality is still supported, so this shouldn't break anything.

Regarding this:
`num_samples=None, bins=None`, I decided to leave flexibility for the implementation to decide on the num_samples if the user defines the bins, and vice versa, but this isn't a strong opinion.

Fixes #4

### CODE REVIEW

File: squigglepy/distributions.py [hunk 1]
```diff
@@ -40,6 +41,14 @@ def __str__(self):
     def __repr__(self):
         return str(self)
 
+    def plot(self, num_samples=None, bins=None):
+        num_samples = 1000 if num_samples is None else num_samples
+        bins = 200 if bins is None else bins
```
Comment (peterhurford) [inline squigglepy/distributions.py:46]: Why not use `def plot(self, num_samples=1000, bins=200)`?

File: README.md [hunk 2]
```diff
@@ -38,6 +38,9 @@ pprint(sq.get_percentiles(samples, digits=0))
 # Histogram
 plt.hist(samples, bins=200)
 plt.show()
+
+# Shorter histogram
+total_tuners_in_2022.plot()
```
Comment (peterhurford) [inline README.md:43]: Nice!

**Review by peterhurford — CHANGES_REQUESTED** [review]

**Comment by peterhurford** [comment]
Thank you!

File: squigglepy/distributions.py (hunk 1 above)
Comment (hibukki) [reply squigglepy/distributions.py:46]: Hey!

Two reasons:

# 1)
TL;DR: What I wrote in the bottom of the PR description:
> Regarding `num_samples=None, bins=None`, I decided to leave flexibility for the implementation to decide on the num_samples if the user defines the bins, and vice versa, but this isn't a strong opinion.

Longer: I can imagine that the user might want to get more samples (sacrificing performance), and that we'll have some clever algorithm to also use more bins in that situation, maybe. Or maybe we'll have a clever implementation to use more bins if each bin has too many values, I don't know.
And we can't do any of that if we can't (in `plot()`'s implementation) distinguish between "the user explicitly asked for 200 bins" and "the user doesn't care how many bins to use"

# 2)
Another way I'm thinking about `plot()`'s signature here:

[seaborn.displot()](https://seaborn.pydata.org/generated/seaborn.displot.html#seaborn.displot) seem to be a best practice for creating an API similar to what I'm trying to make here (according to my DS friend), and they've put much more thought into this than I have, and they decided to make most of their params `None` by default.
image

I can also imagine `BaseDistribution.plot()` getting a similar API to seaborn, and I want to start with an API that will "fit in", if I can, though this is secondary.

---

I'm also ok with making the change as you suggested. This isn't a strong opinion, I'm just sharing my considerations as you asked.

Comment (peterhurford) [reply squigglepy/distributions.py:46]: Ok that's sensible. Thanks!
